synapse/handlers/oidc_handler.py

```diff
@@ -1084,7 +1116,7 @@ def get_remote_user_id(self, userinfo: UserInfo) -> str:
         return userinfo[self._config.subject_claim]
 
     async def map_user_attributes(
-        self, userinfo: UserInfo, token: Token
+        self, userinfo: UserInfo, token: Token, failures: int
```

Comment:
So this just returns the same localpart each time? That sounds inefficient if we have to do that 1000 times? I guess we should either a) raise in here or b) add a check above that returned localparts changed?

Reply:
Oh. I forgot to update this code to handle this situation. By default we should do the same as SAML, which is append `str(failures) if failures else ""`.